Sources/Run_in_Sandbox/RunInSandbox.ps1

```diff
@@ -383,29 +383,33 @@ switch ($Type) {
 
         $add_parameters.add_click({
                 $Script:Paramaters = $parameters_to_add.Text.ToString()
-                $Script:Startup_Command = "wscript.exe $Full_Startup_Path $Paramaters"
+                $Script:Startup_Command = "wscript.exe $Full_Startup_Path_Quoted $Paramaters"
                 $Form_VBS.close()
             })
 
         $Form_VBS.Add_Closing({
                 $Script:Paramaters = $parameters_to_add.Text.ToString()
-                $Script:Startup_Command = "wscript.exe $Full_Startup_Path $Paramaters"
+                $Script:Startup_Command = "wscript.exe $Full_Startup_Path_Quoted $Paramaters"
             })
 
         $Form_VBS.ShowDialog() | Out-Null
 
         New-WSB -Command_to_Run $Startup_Command
     }
     "ZIP" {
-        $Script:Startup_Command = $PSRun_Command + " " + "Expand-Archive $Full_Startup_Path $Sandbox_Desktop_Path\ZIP_extracted"
+        $command = "Expand-Archive -LiteralPath '$Full_Startup_Path' -DestinationPath '$Sandbox_Desktop_Path\ZIP_extracted'"
+        $bytes = [System.Text.Encoding]::Unicode.GetBytes($command)
+        $encodedCommand = [Convert]::ToBase64String($bytes)
```

Comment:
Could you explain a bit more, why you are encoding the command first to base64 and append it to the startup_command in the end instead of appending the command directly to the end of startup_command? What benfits does this have? That way the actual command can no longer be seen in the generated wsb file

Reply:
I **thought** that I had previously found a case for the base64 encoding being required - but after testing again now with a few different ZIP files, I can't find any instance where just using -LiteralPath and wrapping in single quotes isn't enough to make it work. So I suspect that I had originally added the base64 solution before making those other changes, which then made the original solution redundant.

Also, fair point about the command no longer being visible in the WSB.

I have just pushed a new commit to revert this (but still keeping the -LiteralPath and single-quote wrapping tweaks).